src/challenge-status/challenge-status.service.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@nestjs/common';
+import { State } from './entities/state.enum';
+import { ChallengeStatusEntity as ChallengeStatus } from './entities/challenge-status.entity';
+
+@Injectable()
+export class ChallengeStatusService {
+  /**
+   * Creates a challenge status for locked na hidden statuses
+   * @param studentId a student id
+   * @param challenge a challenge student has taken up.
+   */
+  async createStatus(studentId: string, challengeId: string, state: State): Promise<ChallengeStatus> {
+    const status: ChallengeStatus = {
+      startedAt: new Date(),
+      endedAt: undefined,
+      state: [state],
+      openedAt: undefined,
+    };
+    return status;
+  }
+
+  async markAsOpen(studentId: string, challengeId: string, startedDate: Date, date: Date): Promise<ChallengeStatus> {
+    const status: ChallengeStatus = {
```

Comment:
Have a repository for `ChallengeStatus` and query it based on parameters `(studentId, challengeId)`, then update.